src/main/java/prefs/Preferences.java

```diff
@@ -11,122 +17,262 @@
+        FileHelper configFileHelper = new FileHelper(configDirectory, configFileName);
+        String fileContents;
+        if (!configFileHelper.exists()) {
+            fileContents = DEFAULT_FILE_CONTENTS;
+        } else {
+            try {
+                fileContents = configFileHelper.loadFileContents();
+            } catch (IOException e) {
+                // if we can't read the file, just ignore the existing file
+                HTLog.error(logger, e);
+                logger.error(e.toString());
+                fileContents = DEFAULT_FILE_CONTENTS;
+            }
+        }
 
-    public void saveGlobalConfig() {
-        fileHandler.saveGlobalConfig(global);
+        return createConfig(fileContents, configClass);
     }
 
-    private void initGlobalConfig() {
-        global = fileHandler.initGlobalConfig();
+    /**
+     * Saves the session and user configs to file
+     */
+    private void save() {
+        try {
+            saveConfig(sessionConfig, configDirectory, sessionConfigFileName, SessionConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save session config");
+        }
+
+        try {
+            saveConfig(userConfig, configDirectory, userConfigFileName, UserConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save user config");
+        }
     }
 
-    private void loadGlobalConfig() {
-        global = fileHandler.loadGlobalConfig();
+    /**
+     * Saves a Config to file
+     * @param config The config object to save.
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The file name of the config file on disk
+     * @param configClass The class of the config object
+     */
+    private void saveConfig(Config config,
+                            String configDirectory,
+                            String configFileName,
+                            Class configClass) throws IOException {
+        JsonHelper jsonHelper = new JsonHelper();
+        String jsonString = jsonHelper.toJsonString(config, configClass);
+        FileHelper fileHelper = new FileHelper(configDirectory, configFileName);
+        fileHelper.writeFileContents(jsonString);
     }
 
-    // Last login credentials. While the main UI is running (i.e. logged in successfully), last login
-    // credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
-    // being called immediately after a successful login in LoginDialog.
+    /**
+     * Last login credentials. While the main UI is running (i.e. logged in successfully), last login
+     * credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
+     * being called immediately after a successful login in LoginDialog.
+     */
```

Comment:
Can we just update the comment here? Maybe `Retrieves last login password`? Since credentials is both username and password.

Also, the second part of the comment should be removed since this class should have no knowledge of UI class and when the values are set.